http/src/main/java/com/hazelcast/jet/contrib/http/impl/HttpListenerSinkContext.java

```diff
@@ -49,10 +49,10 @@
 
     private final SinkHttpHandler sinkHttpHandler;
     private final Undertow undertow;
-    private final ArrayList<String> messageBuffer;
     private final FunctionEx<T, String> toStringFn;
     private final Ringbuffer<String> ringBuffer;
     private final int accumulateLimit;
+    private final ConcurrentLinkedQueue<String> messageBuffer;
 
     public HttpListenerSinkContext(
```

Comment:
Can't comment to lines not changed in this PR, so doing it here.

1. Isn't there a race in `receive` fn: we check if there's a client connected and either send directly or put to buffer. But what if a client disconnects in the meantime, will that item be lost?

2. Do we need the `flush` method? It's called immediately after `receive` and will do some work only if there wasn't a client and one connected in the meantime, but that's handled in the callback.

Reply:
1. Yes, there is a race there, the real problem is undertow does not understand that a client disconnected without trying to send some items to that client.

2. removed it